native/src/__tests__/Navigator.spec.tsx

```diff
@@ -129,39 +139,50 @@ describe('Navigator', () => {
     jest.clearAllMocks()
   })
 
-  it('should preload cities', async () => {
-    const { findByText } = renderNavigator()
-
-    await findByText('Intro')
-    expect(useLoadCities).toHaveBeenCalled()
-  })
+  const mockSettings = (settings: Partial<SettingsType>) =>
+    mocked<(_: never) => ReturnType<SettingsType>>(useLoadAsync).mockImplementation(() => ({
+      data: { ...defaultSettings, ...settings },
+      error: null,
+      loading: false,
+      refresh: jest.fn(),
+    }))
 
   it('should display categories if a city is selected and the intro was shown', async () => {
-    await appSettings.setIntroShown()
+    mockSettings({ introShown: true })
     const { findByText } = renderNavigator('augsburg')
     await findByText('Categories')
   })
 
+  it('should display landing if the selected city is not available anymore', async () => {
+    mockSettings({ introShown: true })
+    const { findByText } = renderNavigator('disabledCity')
+    await findByText('Landing')
+    expect(changeCityCode).toHaveBeenCalledTimes(1)
+    expect(changeCityCode).toHaveBeenCalledWith(null)
+    expect(dataContainer.deleteCity).toHaveBeenCalledTimes(1)
+    expect(dataContainer.deleteCity).toHaveBeenCalledWith('disabledCity')
+  })
+
   it('should display Landing if no city is selected in settings and intro was shown', async () => {
-    await appSettings.clearSelectedCity()
-    await appSettings.setIntroShown()
+    mockSettings({ introShown: true })
     const { findByText } = renderNavigator()
     await findByText('Landing')
   })
 
   it('should display Intro if intro was not shown yet', async () => {
+    mockSettings({ introShown: false })
     const { findByText } = renderNavigator()
     await findByText('Intro')
   })
 
   it('should listen for push notification press in quit state', async () => {
+    mockSettings({ introShown: true })
     mocked(quitAppStatePushNotificationListener).mockImplementation(async navigate =>
       navigate('https://integreat.app/augsbug/de/news/local/1234')
     )
-    await appSettings.setIntroShown()
-    const { findByText } = renderNavigator()
+    const { getByText } = renderNavigator()
 
-    await findByText('Redirect')
+    waitForExpect(() => expect(getByText('Redirect')).toBeTruthy())
```

Comment:
findByText and waitForExpect should do the same: https://github.com/testing-library/dom-testing-library/blob/d09b3c25b07486c17603317f4c5a0320c188cbed/src/query-helpers.ts#LL120C1-L120C1 If this is making a difference it's because you are missing thi await

Reply:
Thanks for the hint :+1: